### PR TITLE
Jdbc to jdbc

### PR DESCRIPTION
**Pull request by Naveenkm13**

### CODE REVIEW

**Comment by shashank-google** [comment]
@Naveenkm13 can you please test if it works with all 3 flavors of Cloud SQL ?

File: python/dataproc_templates/jdbc/jdbc_to_jdbc.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        if ((partition_parameters != "") & ((input_jdbc_partitioncolumn == "") | (input_jdbc_lowerbound == "") | (input_jdbc_upperbound == ""))):
+            logger.error("Set all the sql partitioning parameters together-jdbctojdbc.input.partitioncolumn,jdbctojdbc.input.lowerbound,jdbctojdbc.input.upperbound. Refer to README.md for more instructions.")
+            exit (1)
+        elif (partition_parameters == ""):
+            input_data=spark.read \
+                .format(constants.FORMAT_JDBC) \
+                .option(constants.JDBC_URL, input_jdbc_url) \
+                .option(constants.JDBC_DRIVER, input_jdbc_driver) \
+                .option(constants.JDBC_TABLE, input_jdbc_table) \
+                .option(constants.JDBC_NUMPARTITIONS, jdbc_numpartitions) \
```
Comment (shashank-google) [inline python/dataproc_templates/jdbc/jdbc_to_jdbc.py:177]: Please verify if this block works, if numpartition > 1
Even if it does, what is the purpose of specifying partitions number without rest of partition parameters?

File: python/dataproc_templates/jdbc/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+## Arguments
+
+* `jdbctojdbc.input.url`: JDBC input URL
+* `jdbctojdbc.input.driver`: JDBC input driver name
+* `jdbctojdbc.input.table`: JDBC input table name
+* `jdbctojdbc.input.partitioncolumn`: JDBC input table partition column name
+* `jdbctojdbc.input.lowerbound`: JDBC input table partition column lower bound which is used to decide the partition stride
+* `jdbctojdbc.input.upperbound`: JDBC input table partition column upper bound which is used to decide the partition stride
+* `jdbctojdbc.numpartitions`: The maximum number of partitions that can be used for parallelism in table reading and writing. Default set to 10
```
Comment (shashank-google) [inline python/dataproc_templates/jdbc/README.md:13]: Also specify that same value will be used for both input and output jdbc connections.

**Review by shashank-google — COMMENTED** [review]
Minor comments.
I am not python expert so
Pablo / Nilo / Surjit need to review.

File: python/dataproc_templates/jdbc/jdbc_to_jdbc.py (hunk 1 above)
Comment (Naveenkm13) [reply python/dataproc_templates/jdbc/jdbc_to_jdbc.py:177]: @shashank-google  yes it works even without the other parameters.  this parameter is used for parallelism in table reading and writing(concurrent JDBC connections). Other parameters are used while reading only, but numpartitions are used for both read and write for parallelism. 

I reverified from the documentation and included these parameters. 
https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html

File: python/dataproc_templates/jdbc/README.md (hunk 2 above)
Comment (Naveenkm13) [reply python/dataproc_templates/jdbc/README.md]: Updated the readme file.

**Comment by Naveenkm13** [comment]
> @Naveenkm13 can you please test if it works with all 3 flavors of Cloud SQL ?

@shashank-google  i tested with below scenarios and it worked. 
- mysql to mysql
- postgresql to postgresql
- SQL server to SQL server
- mysql to postgresql
- postgresql to SQL server

File: python/dataproc_templates/jdbc/README.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
+# 1. JDBC To JDBC
+
+Template for reading data from JDBC table and writing them to a JDBC table. It supports reading partition tabels and write into partitioned or non-partitioned tables.
+
+## Arguments
+
+* `jdbctojdbc.input.url`: JDBC input URL
+* `jdbctojdbc.input.driver`: JDBC input driver name
+* `jdbctojdbc.input.table`: JDBC input table name
+* `jdbctojdbc.input.partitioncolumn`: JDBC input table partition column name
```
Comment (shashank-google) [inline python/dataproc_templates/jdbc/README.md:10]: specify the optional arguments (like partitioncolumn, lower bound etc) in readme

File: python/dataproc_templates/jdbc/README.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    jdbctojdbc.output.create_table.option="PARTITION BY RANGE(id)  (PARTITION p0 VALUES LESS THAN (5),PARTITION p1 VALUES LESS THAN (10),PARTITION p2 VALUES LESS THAN (15),PARTITION p3 VALUES LESS THAN MAXVALUE)"
+    ```
+    * PostgreSQL
+    ```
+    jdbctojdbc.output.create_table.option="PARTITION BY RANGE(id);CREATE TABLE po0 PARTITION OF <table_name> FOR VALUES FROM (MINVALUE) TO (5);CREATE TABLE po1 PARTITION OF <table_name> FOR VALUES FROM (5) TO (10);CREATE TABLE po2 PARTITION OF <table_name> FOR VALUES FROM (10) TO (15);CREATE TABLE po3 PARTITION OF <table_name> FOR VALUES FROM (15) TO (MAXVALUE);"
+    ```
+
+* Additional execution details [refer spark jdbc doc](https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html)
+
+## Example execution:
```
Comment (shashank-google) [inline python/dataproc_templates/jdbc/README.md:125]: in the example execution please use absolute values.
Refer [here](https://github.com/GoogleCloudPlatform/dataproc-templates/blob/main/java/src/main/java/com/google/cloud/dataproc/templates/jdbc/README.md#2-jdbc-to-gcs)

**Review by shashank-google — APPROVED** [review]
minor readme related comments

File: python/dataproc_templates/jdbc/README.md (hunk 4 above)
Comment (Naveenkm13) [reply python/dataproc_templates/jdbc/README.md]: Updated with Absolute values and added for :
- mysql to mysql
- postgresql to postgresql
- SQL server to SQL server
- mysql to postgresql
- mysql to SQL server

**Pull request merged**